test/foo/lib/config/test_configloader.py

```diff
@@ -0,0 +1,55 @@
+from unittest.mock import Mock
+
+import pytest
+
+from foo.lib.config import ConfigLoader, ConfigLoaderException
+
+
+def reset_config_loader_singleton():
```

Comment:
If you decide to stick with singletons you can try to use fixture that runs code automatically before every test
I think it should work:
```suggestion
@pytest.fixture(autouse=True)
def reset_config_loader_singleton():
```